mlrun/feature_store/api.py

```diff
@@ -181,7 +184,15 @@ def ingest(
     """
     if featureset:
         if isinstance(featureset, str):
-            featureset = get_feature_set_by_uri(featureset)
+            # need to strip store prefix from the uri
+            _, stripped_name = parse_store_uri(featureset)
+            try:
+                featureset = get_feature_set_by_uri(stripped_name)
+            except RunDBError as exc:
+                log_and_raise(
+                    reason=f"{exc}. Make sure feature set is saved in db (call ingest)"
+                )
+
```

Comment:
```suggestion
                # TODO: this handling is needed because the generic httpdb error handling doesn't raise the correct error class and doesn't propagate the correct message, until it solved we're manually handling this case to give better user experience, remove this when the error handling is fixed. 
                log_and_raise(
                    reason=f"{exc}. Make sure the feature set is saved in DB (call feature_set.save())"
                )

```